geo-types/src/lib.rs

```diff
@@ -79,6 +80,7 @@ pub use crate::rect::{InvalidRectCoordinatesError, Rect};
 #[macro_use]
 mod macros;
 
+#[cfg(feature = "rstar")]
```

Comment:
Outside of tests, approx is only used in `private_utils`.

I've now gated the inclusion of the private_utils module on rstar integration since that's the only context in which the methods were used.

It's worth mentioning, the methods are used in `geo`, but `geo` enables the rstar feature, so this doesn't break geo.